src/main/java/org/jenkinsci/plugins/workflow/actions/ExecutorTaskInfoAction.java

```diff
@@ -0,0 +1,88 @@
+package org.jenkinsci.plugins.workflow.actions;
+
+import hudson.model.InvisibleAction;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+/**
+ * Records information for a {@code node} block.
+ */
+public class ExecutorTaskInfoAction extends InvisibleAction implements PersistentAction {
```

Comment:
Oooooh, now I remember why I completely avoided overloading `WorkspaceAction` - it has `@Nonnull` on `getNode()`, `getPath()`, and `getLabels()`. So we'd be breaking contracts if we added new methods for "am I still queued?" and returned `null` on `getNode()` and `getPath()` until we're actually on the agent (`getLabels()` could just return an empty `Set`, so that's not a problem).